### DISCUSSION TITLE
How is this constructor supposed to be used?

### DISCUSSION THREAD

**Discussion by claireannecotter** [nextgenhealthcare/connect#5398] [Q&A]
https://github.com/nextgenhealthcare/connect/blob/e66d1122dbeebfcf5a8939beb93d720b2d5071fb/server/src/com/mirth/connect/server/userutil/AlertSender.java#L33

I'm trying to log an error within my java program. I planned to use [AlertSender] with the String channelId constructor. However, it can't seem to find that method? What else do you recommend? How is this supposed to work?

**Comment by jonbartels** [comment]
Show your code please.
Show any errors thrown please.
Use code blocks (three backticks) to format the code and errors.

> How is this supposed to work?

Construct an alert sender for your channel ID then call `sendAlert`. I think that would then hit the alerting configuration and decide which alerts to actually transmit.

I see a place for `AlertSender` but my initial instinct would be to simply `throw` and cause the channel to error then let the natural alert mechanism report on that error.
